src/client/datascience/jupyter/kernels/codeExecution.ts

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { nbformat } from '@jupyterlab/coreutils';
+import { IDisposable, IDisposableRegistry } from '../../../common/types';
+import { noop } from '../../../common/utils/misc';
+import { IDataScienceErrorHandler, IJupyterSession, INotebook } from '../../types';
+import { disposeAllDisposables } from '../../../common/helpers';
+import { createDeferred } from '../../../common/utils/async';
+import * as uuid from 'uuid/v4';
+
+export class CodeExecutionFactory {
+    constructor(
+        private readonly errorHandler: IDataScienceErrorHandler,
+        private readonly disposables: IDisposableRegistry
+    ) {}
+
+    public create(code: string) {
+        // eslint-disable-next-line @typescript-eslint/no-use-before-define
+        return CodeExecution.fromCode(code, this.errorHandler, this.disposables);
+    }
+}
+
+/**
+ * Responsible for execution of an individual cell and manages the state of the cell as it progresses through the execution phases.
```

Comment:
This is picky, but that's kinda how I am 😆 . The comments here look copied over from the other class, they should be updated for the new class (i.e. this is not responsible for execution of an individual cell).